ecommerce/PaymentGateways/GatewayFactory.php

```diff
@@ -0,0 +1,40 @@
+<?php
+/**
+ * Payment gateway factory class
+ *
+ * @package Tutor\Ecommerce
+ * @author Themeum
+ * @link https://themeum.com
+ * @since 3.0.0
+ */
+
+namespace Tutor\PaymentGateways;
+
+/**
+ * Create object of a payment gateway
+ */
+abstract class GatewayFactory {
+
+	/**
+	 * Create an instance of the specified payment gateway.
+	 *
+	 * @param string $gateway The fully qualified class name of the gateway.
+	 * @return GatewayBase
+	 * @throws \InvalidArgumentException If the class does not exist or is not an instance of GatewayBase.
+	 */
+	public static function create( string $gateway ): GatewayBase {
+		if ( ! class_exists( $gateway ) ) {
+			throw new \InvalidArgumentException( "Gateway class {$gateway} does not exist." );
+		}
+
+		$obj = new $gateway();
```

Comment:
Use meaningful variable naming. You can use `$object` or as it is an instance of a class so `$instance`

Reply:
Already used `$obj` name, isn't it meaningful? There are many instances of using short names like: 
max instead of maximum 
min instead of minimum